modules/core/src/test/java/org/apache/ignite/internal/metric/SystemViewSelfTest.java

```diff
@@ -0,0 +1,400 @@
+            });
+
+            barrier.await();
+
+            assertEquals(1, tasks.size());
+
+            ComputeTaskView t = tasks.iterator().next();
+
+            assertFalse(t.internal());
+            assertNull(t.affinityCacheName());
+            assertEquals(-1, t.affinityPartitionId());
+            assertTrue(t.taskClassName().startsWith(getClass().getName()));
+            assertTrue(t.taskName().startsWith(getClass().getName()));
+            assertEquals(g1.localNode().id(), t.taskNodeId());
+            assertEquals("0", t.userVersion());
+
+            barrier.await();
+        }
+    }
+
+    /**
+     * Tests work of {@link SystemView} for compute grid
+     * {@link IgniteCompute#affinityCallAsync(String, Object, IgniteCallable)} call.
+     */
+    @Test
+    public void testComputeAffinityCall() throws Exception {
+        CyclicBarrier barrier = new CyclicBarrier(2);
+
+        try (IgniteEx g1 = startGrid(0)) {
+            SystemView<ComputeTaskView> tasks = g1.context().systemView().view(TASKS_VIEW);
+
+            IgniteCache<Integer, Integer> cache = g1.createCache("test-cache");
+
+            cache.put(1, 1);
+
+            g1.compute().affinityCallAsync("test-cache", 1, () -> {
+                try {
+                    barrier.await();
+                    barrier.await();
+                }
+                catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+
+                return 0;
+            });
+
+            barrier.await();
+
+            assertEquals(1, tasks.size());
+
+            ComputeTaskView t = tasks.iterator().next();
+
+            assertFalse(t.internal());
+            assertEquals("test-cache", t.affinityCacheName());
+            assertEquals(1, t.affinityPartitionId());
+            assertTrue(t.taskClassName().startsWith(getClass().getName()));
+            assertTrue(t.taskName().startsWith(getClass().getName()));
+            assertEquals(g1.localNode().id(), t.taskNodeId());
+            assertEquals("0", t.userVersion());
+
+            barrier.await();
+        }
+    }
+
+    /** */
+    @Test
+    public void testComputeTask() throws Exception {
+        CyclicBarrier barrier = new CyclicBarrier(2);
+
+        try (IgniteEx g1 = startGrid(0)) {
+            SystemView<ComputeTaskView> tasks = g1.context().systemView().view(TASKS_VIEW);
+
+            IgniteCache<Integer, Integer> cache = g1.createCache("test-cache");
+
+            cache.put(1, 1);
+
+            g1.compute().executeAsync(new ComputeTask<Object, Object>() {
+                @Override public @NotNull Map<? extends ComputeJob, ClusterNode> map(List<ClusterNode> subgrid,
+                    @Nullable Object arg) throws IgniteException {
+                    return Collections.singletonMap(new ComputeJob() {
+                        @Override public void cancel() {
+                            System.out.println("SystemViewSelfTest.cancel");
```

Comment:
Some debug output, perhaps can be removed

Reply:
Sorry for that. Fixed.